src/compiler.ts

```diff
@@ -3648,42 +3648,28 @@ export class Compiler extends DiagnosticEmitter {
     // int to float
     } else if (fromType.isIntegerValue && toType.isFloatValue) {
 
+      if (fromType.isShortIntegerValue) expr = this.ensureSmallIntegerWrap(expr, fromType); // must clear garbage bits
```

Comment:
```suggestion
      // Clear extra bits.
      expr = this.ensureSmallIntegerWrap(expr, fromType);
```

Reply:
Why we need to remove condition here?

Reply:
`ensureSmallIntegerWrap` does the check for you.